### PR TITLE
Improve Docs for Demos

### PR DESCRIPTION
**Pull request by Eric-Wallace-WebHost**
For the new interpretation changes, we'd like to add some nicer docs to make it easy to add new models and interpretation methods.

I'm not sure the best way to go about this. But this PR deletes two files that are out of date and adds a file that sums everything up on adding a new demo.

### CODE REVIEW

**Comment by schmmd** [comment]
I'm all for moving docs to the root.  I wouldn't have found docs under `demo/` easily.

Our current readme is presently very short--so maybe that does make sense to additionally combine `CONTRIBUTING.md` and `README.md`, but I don't have a strong opinion here.

**Comment by Eric-Wallace-WebHost** [comment]
I moved everything into the README.md

File: README.md [hunk 1]
```diff
@@ -1,21 +1,10 @@
...
-
-```
-$ export GIT_HASH=`git log -1 --pretty=format:"%H"`
-$ docker build -t allennlp/demo:$GIT_HASH .
-$ mkdir -p $HOME/.allennlp
-$ docker run -p 8000:8000 -v $HOME/.allennlp:/root/.allennlp --rm allennlp/demo:$GIT_HASH
-```
-
-Note that the `run` process may get killed prematurely if there is insufficient memory allocated to Docker. As of September 14, 2018, setting a memory limit of 10GB was sufficient to run the demo. See [Docker Docs](https://docs.docker.com/docker-for-mac/#advanced) for more on setting memory allocation preferences.
+This repository contains the [AllenNLP demo](https://demo.allennlp.org/reading-comprehension).
```
Comment (matt-gardner) [inline README.md]: Remove the trailing `reading-comprehension` in the link.

**Comment by Eric-Wallace-WebHost** [comment]
:+1: mergeable @matt-gardner

**Pull request merged**